src/fugw/solvers/utils.py

```diff
@@ -703,6 +710,146 @@ def solver_mm_sparse(
     )
 
 
+def solver_mm_l2_sparse(
+    cost, init_pi, uot_params, tuple_weights, train_params, verbose=True
+):
+    niters, tol, eval_freq = train_params
```

Comment:
I don't think that this way of reducing the number of params is very useful. AT least, it should be consistent over the code base...

Reply:
I agree, this concatenation of parameters is difficult to understand. It is somewhat consistent across solvers, but I think we should flatten these variables. Once again, I think this would fit more naturally in a follow-up PR.